driver/cat_driver.hpp

```diff
@@ -0,0 +1,341 @@
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_CAT_DRIVER_HPP
+#define GUARD_MIOPEN_CAT_DRIVER_HPP
+
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "random.hpp"
+#include "tensor_driver.hpp"
+#include "timer.hpp"
+#include <algorithm>
+#include <cfloat>
+#include <cstdlib>
+#include <memory>
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include <numeric>
+#include <vector>
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+
+#ifndef MLO_CATHOST_H_
+#define MLO_CATHOST_H_
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloCatForwardRunHost(std::vector<miopenTensorDescriptor_t> inputDescs,
+                             std::vector<Tgpu*> inputs,
+                             miopenTensorDescriptor_t outputDesc,
+                             Tcheck* outputhost,
+                             int32_t dim)
+{
+    auto shape             = miopen::deref(outputDesc).GetLengths();
+    size_t outer_size      = 1;
+    size_t inner_size      = 1;
+    size_t output_dim_size = shape[dim];
+    size_t i               = 0;
+    for(; i < dim; i++)
+    {
+        outer_size *= shape[i];
+    }
+
+    for(; i < shape.size(); i++)
+    {
+        inner_size *= shape[i];
+    }
+
+    int32_t ret                = 0;
+    size_t output_start_offset = 0;
+
+    for(i = 0; i < inputs.size(); i++)
+    {
+        auto input       = inputs[i];
+        size_t dim_size  = miopen::deref(inputDescs[i]).GetLengths()[dim];
+        size_t copy_size = inner_size / output_dim_size * dim_size;
+        for(size_t o = 0; o < outer_size; o++)
+        {
+            size_t output_offset = output_start_offset + (o * inner_size);
+            for(size_t j = 0; j < copy_size; j++)
+            {
+                outputhost[output_offset + j] = input[copy_size * o + j];
+            }
+        }
+        output_start_offset += copy_size;
+    }
+
+    return ret;
+}
+
+#endif
+
+template <typename Tgpu, typename Tref = Tgpu>
+class CatDriver : public Driver
+{
+public:
+    CatDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&outputDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    std::vector<std::vector<int>> GetInputTensorLengthsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~CatDriver() override
+    {
+        for(auto inputDesc : inputDescs)
+        {
+            miopenDestroyTensorDescriptor(inputDesc);
+        }
+        miopenDestroyTensorDescriptor(outputDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+    int dim_size;
```

Comment:
Should :dim" and "dim_size" has unsigned type?

Reply:
change to uint32_t too
and remove unused forw